ejb/src/main/java/biz/karms/sinkit/ejb/impl/ArchiveServiceEJB.java

```diff
@@ -88,13 +88,10 @@ public boolean archiveReceivedIoCRecord(final IoCRecord ioc) throws ArchiveExcep
         //compute uniqueReference
         ioc.setUniqueRef(IoCIdentificationUtils.computeUniqueReference(ioc));
         final Map<String, Map<String, Object>> fieldsToUpdate = new HashMap<>();
-        fieldsToUpdate.put("seen", new HashMap<String, Object>(){
-            {
-                put("last", ioc.getSeen().getLast());
-                // TODO: Isn't this field obsoleted by the new accuracy model?
-                put("accuracy", ioc.getFeed().getAccuracy());
-            }
-        });
+        HashMap<String,Object> lastseen = new HashMap<String, Object>();
+        lastseen.put("last", ioc.getSeen().getLast());
+        fieldsToUpdate.put("seen", lastseen);
```

Comment:
@hugeox Could you describe what is the actual fix here? Wasn't the only bug the accuracy being nested in the "seen" map? THX.

Reply:
Yes, that was indeed a bug but since we now represent accuracy differently it just had to be changed.
But even the updates of "seen.last" didn't work, so I just fixed that by using simpler code (I didn't quite see why the original version wouldn't work though, but it just didn't)